torch/nn/modules/module.py

```diff
@@ -1994,6 +2007,10 @@ def load_state_dict(self, state_dict: Mapping[str, Any],
             strict (bool, optional): whether to strictly enforce that the keys
                 in :attr:`state_dict` match the keys returned by this module's
                 :meth:`~torch.nn.Module.state_dict` function. Default: ``True``
+            assign (bool, optional): whether to assign items in the state
+            dictionary to their corresponding keys in the module instead
+            of copying them inplace into the module's parameters and buffers.
+            Default: ``False``
```

Comment:
I think here might be a good place to have a quick "Warning: if you use this option, you MUST create the optimizer after loading the state dict."